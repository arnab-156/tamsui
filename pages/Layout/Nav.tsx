import React from 'react';
import classNames from 'classnames';
import { Link } from 'react-router-dom';

import styles from './styles.nav.module.scss';

function Nav() {
  return (
    <nav className={styles.navbar}>
      <div className={styles.contents}>
        <div className={classNames(styles.segment, styles.home)}>
          <Link className={styles.menuLink} to="/">
            <img alt="Tamsui logo" className={styles.logo} src="/static/images/logo.webp" />
            <span>Tamsui</span>
          </Link>
        </div>
        <div className={classNames(styles.segment, styles.pages)}>
          <Link className={styles.menuLink} to="/counter">
            Counter
          </Link>
        </div>
        <div className={classNames(styles.segment, styles.social)}>
          <a href="https://github.com/chichiwang/tamsui" rel="noreferrer" target="_blank">
            <img alt="Github logo" className={styles.logo} src="/static/images/github-mark.svg" />
          </a>
        </div>
      </div>
    </nav>
  );
}

export default Nav;
