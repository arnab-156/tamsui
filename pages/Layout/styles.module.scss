@use "sass:map";
@use "app/var.module";

.page {
  width: 100%;
  min-height: 100vh;
}

.content {
  font-size: 1rem;
  max-width: map.get(var.$content, "max-width");
  padding-top: 5rem;
}
