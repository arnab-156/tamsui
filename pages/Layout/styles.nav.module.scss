@use "sass:map";
@use "app/var.module";

$section-border-size: 4px;

.navbar {
  width: 100%;
  font-size: 1.125rem;
}

.contents {
  display: flex;
  max-width: map.get(var.$content, "max-width");
}

.segment {
  display: flex;
  flex-basis: 33%;
  flex-grow: 1;
  align-items: center;
  padding: calc(map.get(var.$content, "padding") / 2) map.get(var.$content, "padding");

  &.home {
    justify-content: flex-start;
    border-bottom: $section-border-size solid map.get(var.$colors, "maroon");

    & span {
      margin: 0 calc(var.$spacing-unit / 2);
      vertical-align: middle;
    }

    & .logo {
      width: auto;
      height: 35px;
    }
  }

  &.pages {
    justify-content: center;
    border-bottom: $section-border-size solid map.get(var.$colors, "red");
  }

  &.social {
    justify-content: flex-end;
    border-bottom: $section-border-size solid map.get(var.$colors, "orange");

    & .logo {
      width: auto;
      height: 30px;
    }
  }

  .menuLink {
    color: map.get(var.$colors, "black");
    font-weight: normal;
  }

  & .logo {
    width: auto;
    height: 35px;
    vertical-align: middle;
  }
}
