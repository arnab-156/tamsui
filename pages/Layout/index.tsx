import React from 'react';

import Nav from './Nav';

import styles from './styles.module.scss';

function Layout({ children }: React.PropsWithChildren) {
  return (
    <section className={styles.page}>
      <Nav />
      <main className={styles.content}>
        {children}
      </main>
    </section>
  );
}

export default Layout;
