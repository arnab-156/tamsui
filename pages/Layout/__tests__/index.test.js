import React from 'react';
import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router-dom';

import Layout from '../index';

function RouterWrappedLayout({ children }) {
  return (
    <MemoryRouter>
      <Layout>
        { children }
      </Layout>
    </MemoryRouter>
  );
}

function ChildComponent() {
  return (
    <h1>Layout Child Component</h1>
  );
}

describe('Layout Component - default', () => {
  test('matches no-child snapshot', () => {
    const tree = renderer.create(<RouterWrappedLayout />).toJSON();

    expect(tree).toMatchSnapshot();
  });

  test('matches child snapshot', () => {
    const tree = renderer.create(
      <RouterWrappedLayout>
        <ChildComponent />
      </RouterWrappedLayout>,
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
