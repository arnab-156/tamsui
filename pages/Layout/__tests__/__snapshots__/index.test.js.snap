// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout Component - default matches child snapshot 1`] = `
<section
  className="page"
>
  <nav
    className="navbar"
  >
    <div
      className="contents"
    >
      <div
        className="segment home"
      >
        <a
          className="menuLink"
          href="/"
          onClick={[Function]}
        >
          <img
            alt="Tamsui logo"
            className="logo"
            src="/static/images/logo.webp"
          />
          <span>
            Tamsui
          </span>
        </a>
      </div>
      <div
        className="segment pages"
      >
        <a
          className="menuLink"
          href="/counter"
          onClick={[Function]}
        >
          Counter
        </a>
      </div>
      <div
        className="segment social"
      >
        <a
          href="https://github.com/chichiwang/tamsui"
          rel="noreferrer"
          target="_blank"
        >
          <img
            alt="Github logo"
            className="logo"
            src="/static/images/github-mark.svg"
          />
        </a>
      </div>
    </div>
  </nav>
  <main
    className="content"
  >
    <h1>
      Layout Child Component
    </h1>
  </main>
</section>
`;

exports[`Layout Component - default matches no-child snapshot 1`] = `
<section
  className="page"
>
  <nav
    className="navbar"
  >
    <div
      className="contents"
    >
      <div
        className="segment home"
      >
        <a
          className="menuLink"
          href="/"
          onClick={[Function]}
        >
          <img
            alt="Tamsui logo"
            className="logo"
            src="/static/images/logo.webp"
          />
          <span>
            Tamsui
          </span>
        </a>
      </div>
      <div
        className="segment pages"
      >
        <a
          className="menuLink"
          href="/counter"
          onClick={[Function]}
        >
          Counter
        </a>
      </div>
      <div
        className="segment social"
      >
        <a
          href="https://github.com/chichiwang/tamsui"
          rel="noreferrer"
          target="_blank"
        >
          <img
            alt="Github logo"
            className="logo"
            src="/static/images/github-mark.svg"
          />
        </a>
      </div>
    </div>
  </nav>
  <main
    className="content"
  />
</section>
`;
